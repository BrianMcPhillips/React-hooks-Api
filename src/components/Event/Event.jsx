import React from 'react';
import PropTypes from 'prop-types';

const Event = ({ year, text }) => (
  <>
    <h3>{year}</h3>
    <p>{text}</p>
  </>
);

Event.propTypes = {
  year: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default Event;
