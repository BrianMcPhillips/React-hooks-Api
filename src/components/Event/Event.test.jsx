import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Event from './Event';

describe('Event component', () => {
  afterEach(() => cleanup());
  it('renders Event', () => {
    const { asFragment } = render(
      <Event 
        text="Best event"
        year="2022"
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
