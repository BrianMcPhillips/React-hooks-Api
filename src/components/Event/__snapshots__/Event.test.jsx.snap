// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event component renders Event 1`] = `
<DocumentFragment>
  <h3>
    2022
  </h3>
  <p>
    Best event
  </p>
</DocumentFragment>
`;
